src/main/java/org/sagebionetworks/bridge/hibernate/MySQLHibernatePersistenceExceptionConverter.java

```diff
@@ -82,15 +95,19 @@ public RuntimeException convert(PersistenceException exception, Object entity) {
         if (throwable instanceof java.sql.SQLIntegrityConstraintViolationException) {
             String rawMessage = throwable.getMessage();
             String displayMessage = DEFAULT_CONSTRAINT_MSG;
-            
             // SQLIntegrityConstraintViolation can contain different constraint violations, 
             // including foreign key and duplicate entry violations
             if (rawMessage.contains("Duplicate entry")) {
                 displayMessage = selectMsg(
                         rawMessage, UNIQUE_KEY_CONSTRAINTS, UNIQUE_CONSTRAINT_MSG, name, displayMessage);
             } else if (rawMessage.contains("a foreign key constraint fails")) {
-                displayMessage = selectMsg(
-                        rawMessage, FOREIGN_KEY_CONSTRAINTS, FK_CONSTRAINT_MSG, name, displayMessage);
+                if (rawMessage.contains("Cannot add or update a child row")) {
+                    displayMessage = selectMsg(
+                            rawMessage, FOREIGN_KEY_MISSING_PARENT_CONSTRAINTS, MISSING_PARENT_INVALID_FK_CONSTRAINT_MSG, name, displayMessage);
+                } else {
+                    displayMessage = selectMsg(
+                            rawMessage, FOREIGN_KEY_CONSTRAINTS, FK_CONSTRAINT_MSG, name, displayMessage);
+                }
             }
```

Comment:
I think instead of nesting these rawMessage string checks, they should be in a flat if/else but ordered correctly (in this case, your message check needs to come before "rawMessage.contains("a foreign key constraint fails")"

Reply:
Changed to flat if/else.